mesa/agent.py

```diff
@@ -36,6 +37,81 @@ def step(self) -> None:
     def advance(self) -> None:
         pass
 
+    def forward_backward(self, amount, sign):
+        """Does the calculation to find  the agent's next move and is used within the forward and backward functions"""
+        new_x = float(self.pos[0]) + sign * math.cos(self.heading * math.pi / 180) * amount
+        new_y = float(self.pos[1]) + sign * math.sin(self.heading * math.pi / -180) * amount
+        next_move = (new_x, new_y)
+        try:
+            self.model.space.move_agent(self, next_move)
+        except:
+            print("agent.py (forward_backwards): could not move agent within self.model.space")
+
+    def forward(self, amount):
+        """Moves the agent forward by the amount given"""
+        self.forward_backward(amount, 1)
+
+    def back(self, amount):
+        """Moves the agent backwards from where its facing by the given amount"""
+        self.forward_backward(amount, -1)
+
+    def right(self, degree):
+        """Turns the agent right by the given degree"""
+        self.heading = (self.heading - degree) % 360
+
+    def left(self, degree):
+        """Turns the agent left by the given degree"""
+        self.heading = (self.heading + degree) % 360
+
+    def setxy(self, x, y):
```

Comment:
What about `set_position(self, position)`? This way, the API remains the same in the implementation with higher number of dimensions, instead of e.g. `set_xyz` for 3D.

Reply:
`setxy` needs to be removed now that we have `set_pos`.